services/tftp/src/main/java/org/openthinclient/tftp/TFTPService.java

```diff
@@ -63,8 +73,37 @@
 
   public void startService() throws Exception {
     try {
-      tftpServer = new TFTPServer(0 != tftpPort
-          ? tftpPort
+      for (TFTPServiceConfiguration.Export export : configuration.getExports()) {
+
+          String prefix = export.getPrefix();
+          String providerClassName = null;
+          if (export.getProviderClass() != null) {
+              providerClassName = export.getProviderClass().getName();
+          }
+          String basedir = export.getBasedir();
+          List<TFTPServiceConfiguration.Export.Option> options = export.getOptions();
+
+          LOGGER.info("Exporting " + prefix + "="
+                  + (null != providerClassName ? providerClassName : basedir)
+                  + " options=" + options);
+
+          if (null != providerClassName) {
+              try {
+                  Map<String, String> opts = new HashMap<String, String>();
+                  for (TFTPServiceConfiguration.Export.Option option : options) {
+                      opts.put(option.getName(), option.getValue());
+                  }
+                  addExport(new TFTPExport(prefix, providerClassName, opts));
```

Comment:
Wouldn't it be better to pass the providerClass instance instead of the name here? This would avoid the try ... catch ... block with ClassNotFoundExceptions, that should never really happen